### ISSUE TITLE
Calico-node pods fails. Goroutines fails with panic.

### ISSUE THREAD

**Issue by stepanselyuk** [projectcalico/calico#5410]
I installed microk8s on Ubuntu 21.10 server 64 bit on Raspberry Pi 4B (2x4Gb, 1x8Gb), as I added three nodes, MicroK8s enabled HA-mode automatically (if that will help). Kubernetes version is 1.22; the initial setup was with docker.io/calico/kube-controllers:v3.17.3 and docker.io/calico/node:v3.19.1 (not sure about the latter); I spotted 2/3 of calico nodes is in a not-ready state with restarts. If I started with k8s-001 node, then calico node fails on 002 and 003. Then I updated every component (controller and daemonset) to use tag `release-v3.21-arm64` but this is not fixed the issue.

## Expected Behavior
Should work fine

## Current Behavior
```
[calico-node-bjlw6 flexvol-driver]  File exists /host/driver/uds. Copy over
[calico-node-bjlw6 install-cni]  time="2022-01-11T21:25:11Z" level=info msg="Running as a Kubernetes pod" source="install.go:141"
[calico-node-bjlw6 install-cni]  time="2022-01-11T21:25:16Z" level=info msg="Installed /host/opt/cni/bin/bandwidth"
[calico-node-bjlw6 install-cni]  time="2022-01-11T21:25:18Z" level=info msg="Installed /host/opt/cni/bin/calico"
[calico-node-bjlw6 install-cni]  time="2022-01-11T21:25:19Z" level=info msg="Installed /host/opt/cni/bin/calico-ipam"
[calico-node-bjlw6 install-cni]  time="2022-01-11T21:25:19Z" level=info msg="Installed /host/opt/cni/bin/flannel"
[calico-node-bjlw6 install-cni]  time="2022-01-11T21:25:19Z" level=info msg="Installed /host/opt/cni/bin/host-local"
[calico-node-bjlw6 install-cni]  time="2022-01-11T21:25:21Z" level=info msg="Installed /host/opt/cni/bin/install"
[calico-node-bjlw6 install-cni]  time="2022-01-11T21:25:21Z" level=info msg="Installed /host/opt/cni/bin/loopback"
[calico-node-bjlw6 install-cni]  time="2022-01-11T21:25:21Z" level=info msg="Installed /host/opt/cni/bin/portmap"
[calico-node-bjlw6 install-cni]  time="2022-01-11T21:25:21Z" level=info msg="Installed /host/opt/cni/bin/tuning"
[calico-node-bjlw6 install-cni]  time="2022-01-11T21:25:21Z" level=info msg="Wrote Calico CNI binaries to /host/opt/cni/bin\n"
[calico-node-bjlw6 install-cni]  time="2022-01-11T21:25:22Z" level=info msg="CNI plugin version: v3.21.2-21726-g873174aa5eec\n"
[calico-node-bjlw6 install-cni]  time="2022-01-11T21:25:22Z" level=info msg="/host/secondary-bin-dir is not writeable, skipping"
[calico-node-bjlw6 install-cni]  time="2022-01-11T21:25:22Z" level=info msg="Using CNI config template from CNI_NETWORK_CONFIG environment variable." source="install.go:322"
[calico-node-bjlw6 install-cni]  {
[calico-node-bjlw6 install-cni]    "name": "k8s-pod-network",
[calico-node-bjlw6 install-cni]    "cniVersion": "0.3.1",
[calico-node-bjlw6 install-cni]    "plugins": [
[calico-node-bjlw6 install-cni]      {
[calico-node-bjlw6 install-cni]        "type": "calico",
[calico-node-bjlw6 install-cni]        "log_level": "info",
[calico-node-bjlw6 install-cni]        "datastore_type": "kubernetes",
[calico-node-bjlw6 install-cni]        "nodename_file_optional": true,
[calico-node-bjlw6 install-cni]        "nodename": "rpi-k8s-003.local",
[calico-node-bjlw6 install-cni]        "mtu": 1440,
[calico-node-bjlw6 install-cni]        "ipam": {
[calico-node-bjlw6 install-cni]            "type": "calico-ipam"
[calico-node-bjlw6 install-cni]        },
[calico-node-bjlw6 install-cni]        "policy": {
[calico-node-bjlw6 install-cni]            "type": "k8s"
[calico-node-bjlw6 install-cni]        },
[calico-node-bjlw6 install-cni]        "kubernetes": {
[calico-node-bjlw6 install-cni]            "kubeconfig": "/var/snap/microk8s/current/args/cni-network/calico-kubeconfig"
[calico-node-bjlw6 install-cni]        }
[calico-node-bjlw6 install-cni]      },
[calico-node-bjlw6 install-cni]      {
[calico-node-bjlw6 install-cni]        "type": "portmap",
[calico-node-bjlw6 install-cni]        "snat": true,
[calico-node-bjlw6 install-cni]        "capabilities": {"portMappings": true}
[calico-node-bjlw6 install-cni]      },
[calico-node-bjlw6 install-cni]      {
[calico-node-bjlw6 install-cni]        "type": "bandwidth",
[calico-node-bjlw6 install-cni]        "capabilities": {"bandwidth": true}
[calico-node-bjlw6 install-cni]      }
[calico-node-bjlw6 install-cni]    ]
[calico-node-bjlw6 install-cni]  }
[calico-node-bjlw6 install-cni]  time="2022-01-11T21:25:22Z" level=info msg="Created /host/etc/cni/net.d/10-calico.conflist"
[calico-node-bjlw6 install-cni]  time="2022-01-11T21:25:22Z" level=info msg="Done configuring CNI.  Sleep= false"
[calico-node-bjlw6 calico-node]  github.com/projectcalico/felix/dataplane/linux.(*vxlanManager).getParentInterface(0x400024e100, 0x4000c2a000, 0x400024e100, 0x0, 0x400024e100, 0x2af21c8)
[calico-node-bjlw6 calico-node]  	/go/pkg/mod/github.com/projectcalico/felix@v0.0.0-20211203211657-ea53c8654697/dataplane/linux/vxlan_mgr.go:480 +0x3c
[calico-node-bjlw6 calico-node]  github.com/projectcalico/felix/dataplane/linux.(*vxlanManager).getLocalVTEPParent(0x400024e100, 0x4000c2a000, 0x1, 0x1, 0x400007cf40)
[calico-node-bjlw6 calico-node]  	/go/pkg/mod/github.com/projectcalico/felix@v0.0.0-20211203211657-ea53c8654697/dataplane/linux/vxlan_mgr.go:274 +0x34
[calico-node-bjlw6 calico-node]  github.com/projectcalico/felix/dataplane/linux.(*vxlanManager).KeepVXLANDeviceInSync(0x400024e100, 0x5a0, 0x29ddc00, 0x2540be400)
[calico-node-bjlw6 calico-node]  	/go/pkg/mod/github.com/projectcalico/felix@v0.0.0-20211203211657-ea53c8654697/dataplane/linux/vxlan_mgr.go:449 +0x214
[calico-node-bjlw6 calico-node]  created by github.com/projectcalico/felix/dataplane/linux.NewIntDataplaneDriver
[calico-node-bjlw6 calico-node]  	/go/pkg/mod/github.com/projectcalico/felix@v0.0.0-20211203211657-ea53c8654697/dataplane/linux/int_dataplane.go:447 +0xcec
[calico-node-bjlw6 calico-node]  W0112 12:33:01.881648     380 warnings.go:70] discovery.k8s.io/v1beta1 EndpointSlice is deprecated in v1.21+, unavailable in v1.25+; use discovery.k8s.io/v1 EndpointSlice
[calico-node-bjlw6 calico-node]  panic: runtime error: invalid memory address or nil pointer dereference
[calico-node-bjlw6 calico-node]  [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1653dc8]
[calico-node-bjlw6 calico-node]  
[calico-node-bjlw6 calico-node]  goroutine 65 [running]:
[calico-node-bjlw6 calico-node]  github.com/vishvananda/netlink.(*Handle).newNetlinkRequest(...)
[calico-node-bjlw6 calico-node]  	/go/pkg/mod/github.com/vishvananda/netlink@v1.1.1-0.20210703095558-21f2c55a7727/handle_linux.go:149
[calico-node-bjlw6 calico-node]  github.com/vishvananda/netlink.(*Handle).LinkList(0x0, 0x0, 0x4000078c78, 0x5471e8, 0x48, 0x286c200)
[calico-node-bjlw6 calico-node]  	/go/pkg/mod/github.com/vishvananda/netlink@v1.1.1-0.20210703095558-21f2c55a7727/link_linux.go:2008 +0x28
[calico-node-bjlw6 calico-node]  github.com/projectcalico/felix/dataplane/linux.(*vxlanManager).getParentInterface(0x400049a680, 0x40009160c0, 0x400049a680, 0x0, 0x400049a680, 0x2af21c8)
[calico-node-bjlw6 calico-node]  	/go/pkg/mod/github.com/projectcalico/felix@v0.0.0-20211203211657-ea53c8654697/dataplane/linux/vxlan_mgr.go:480 +0x3c
[calico-node-bjlw6 calico-node]  github.com/projectcalico/felix/dataplane/linux.(*vxlanManager).getLocalVTEPParent(0x400049a680, 0x40009160c0, 0x1, 0x1, 0x4000078f40)
[calico-node-bjlw6 calico-node]  	/go/pkg/mod/github.com/projectcalico/felix@v0.0.0-20211203211657-ea53c8654697/dataplane/linux/vxlan_mgr.go:274 +0x34
[calico-node-bjlw6 calico-node]  github.com/projectcalico/felix/dataplane/linux.(*vxlanManager).KeepVXLANDeviceInSync(0x400049a680, 0x5a0, 0x29ddc00, 0x2540be400)
[calico-node-bjlw6 calico-node]  	/go/pkg/mod/github.com/projectcalico/felix@v0.0.0-20211203211657-ea53c8654697/dataplane/linux/vxlan_mgr.go:449 +0x214
[calico-node-bjlw6 calico-node]  created by github.com/projectcalico/felix/dataplane/linux.NewIntDataplaneDriver
[calico-node-bjlw6 calico-node]  	/go/pkg/mod/github.com/projectcalico/felix@v0.0.0-20211203211657-ea53c8654697/dataplane/linux/int_dataplane.go:447 +0xcec
[calico-node-bjlw6 upgrade-ipam]  2022-01-11 21:25:08.806 [INFO][1] ipam_plugin.go 75: migrating from host-local to calico-ipam...
[calico-node-bjlw6 upgrade-ipam]  2022-01-11 21:25:08.822 [INFO][1] migrate.go 67: checking host-local IPAM data dir dir existence...
[calico-node-bjlw6 upgrade-ipam]  2022-01-11 21:25:08.823 [INFO][1] migrate.go 69: host-local IPAM data dir dir not found; no migration necessary, successfully exiting...
[calico-node-bjlw6 upgrade-ipam]  2022-01-11 21:25:08.823 [INFO][1] ipam_plugin.go 105: migration from host-local to calico-ipam complete node="rpi-k8s-003.local"
[calico-node-bjlw6 calico-node]  2022-01-12 12:33:04.629 [INFO][92] status-reporter/watchercache.go 175: Full resync is required ListRoot="/calico/resources/v3/projectcalico.org/caliconodestatuses"
[calico-node-bjlw6 calico-node]  W0112 12:33:04.675933     412 warnings.go:70] discovery.k8s.io/v1beta1 EndpointSlice is deprecated in v1.21+, unavailable in v1.25+; use discovery.k8s.io/v1 EndpointSlice
[calico-node-bjlw6 calico-node]  W0112 12:33:04.781015     412 warnings.go:70] discovery.k8s.io/v1beta1 EndpointSlice is deprecated in v1.21+, unavailable in v1.25+; use discovery.k8s.io/v1 EndpointSlice
[calico-node-bjlw6 calico-node]  panic: runtime error: invalid memory address or nil pointer dereference
[calico-node-bjlw6 calico-node]  [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1653dc8]
```

## Steps to Reproduce (for bugs)

1. Install Ubuntu 64bit Server 21.10 on Raspberry Pi 4B and Microk8s 1.22 via Snap
2. Add another two nodes to microk8s

## Context
Trying to make a home-based cluster for small projects

## Your Environment
* Calico version: 3.21
* Orchestrator version (e.g. kubernetes, mesos, rkt): kubernetes via microk8s
* Operating System and version: Ubuntu Server 21.10 64bit

**Comment by cguertin14** [comment]
Same problem, running on ARM.

**Comment by caseydavenport** [comment]
https://github.com/vishvananda/netlink/issues/712#issuecomment-1011334108

From the netlink issue, sounds like this is likely due to a missing kernel module (though not clear which)

**Comment by stepanselyuk** [comment]
I remember also the `host-access` plugin for microk8s did a few unexpected things / network issues and when I disabled it - microk8s on RPi4B started to work fine.

**Comment by fafrd** [comment]
Solved for me on Ubuntu 22.04 ARM with `sudo apt install linux-modules-extra-raspi`
